django/contrib/contenttypes/fields.py

```diff
@@ -25,6 +25,20 @@
 from django.utils.functional import cached_property
 
 
+def serialize_pk(obj):
+    opts = obj._meta
+    if opts.is_composite_pk:
+        return opts.pk.to_json(obj.pk)
+    return obj.pk
+
+
+def deserialize_pk(pk, ct):
+    opts = ct.model_class()._meta
+    if opts.is_composite_pk:
+        return opts.pk.from_json(pk)
+    return pk
```

Comment:
Since 6a1a9c0eade674780060cf8af5f5b3375156cdd5 is merged, can you look into using `value_to_string` and `to_python`?

Reply:
Yes. I think we can ship this in the next release, let's focus on resolving the blockers with the CompositePrimaryKey field for 5.2

Reply:
Agreed :star:

Reply:
Small note that work [on this PR](https://github.com/django/django/pull/19273/files#r1996978311) might get us closer to functional composite gfk prefetch.